autogpt_platform/backend/backend/executor/utils.py

```
@@ -1094,7 +1094,19 @@ async def stop_graph_execution(
                 return_exceptions=True,  # Don't fail parent stop if child stop fails
             )
 
-    # Now stop this execution
+    # REL-002 durable cancellation: persist intent before fanout so
+    # executor restart cannot erase it. The column is the SoR; fanout is
+    # latency optimization only.
+    try:
+        from backend.data.execution import set_cancel_requested
+
+        await set_cancel_requested(graph_exec_id, user_id)
+    except Exception as e:
```

Ruff (0.16.3)

[warning] Do not catch blind exception: `Exception`

(BLE001)